war/src/main/java/io/lumeer/engine/controller/CollectionMetadataFacade.java

```diff
@@ -342,16 +344,24 @@ public void renameAttribute(String collectionCode, String oldFullName, String ne
     *       attribute to be dropped
     */
    public void dropAttribute(String collectionCode, String attributeName) {
-      DataDocument removeAttributes = dataStorage.readDocumentIncludeAttrs(metadataCollection(),
-            dialect.combineFilters(collectionCodeFilter(collectionCode), attributeWildcardFilter(collectionCode, attributeName)),
-            Collections.singletonList(LumeerConst.Collection.ATTRIBUTES));
+      DataFilter filter = collectionCodeFilter(collectionCode);
+      List<String> attributes = Collections.singletonList(LumeerConst.Collection.ATTRIBUTES);
 
-      if (removeAttributes == null) {
+      DataDocument collectionToRemoveFrom = dataStorage.readDocumentIncludeAttrs(metadataCollection(), filter, attributes);
+
+      if (collectionToRemoveFrom == null) {
          return;
       }
 
-      dataStorage.removeItemsFromArray(metadataCollection(), collectionCodeFilter(collectionCode), LumeerConst.Collection.ATTRIBUTES,
-            removeAttributes.getArrayList(LumeerConst.Collection.ATTRIBUTES, DataDocument.class));
+      // match "<attributeName>", or "<attributeName>.<child>"
+      Pattern pattern = Pattern.compile(Pattern.quote(attributeName) + "(\\..*)?");
+      Matcher matcher = pattern.matcher("");
+
```

Comment:
And what about dropping attribute `(<something>)*.<attributeName>.<child>` ? Old code considered also this situations, but I think that now it's not working like that.  Or am I wrong?